src/main/webapp/dev/js/common/instructor.js

```diff
@@ -358,14 +358,18 @@ function bindCourseDeleteAllLinks() {
         event.preventDefault();
 
         const $clickedLink = $(event.currentTarget);
-        const messageText = 'Are you sure you want to permanently delete all the courses? '
-                + 'This operation will delete all students and sessions in these courses. '
-                + 'All instructors of these courses will not be able to access them hereafter as well.';
+        const softDeletedCourses = $('[data-soft-deleted-course-id]').toArray();
```

Comment:
I think there is a misunderstanding here. Selecting by `data-*` attributes like that is not a very good idea. Rather, you should continue to select by `id` and call `.data('softDeletedCourseId')` appropriately.